### ISSUE TITLE
beat messages never arriving to rabbitmq

### ISSUE THREAD

**Issue by pimlinders** [celery/celery#2618]
I have set up a beat schedule in django settings file. I have the following perdiodic task

```
CELERYBEAT_SCHEDULE = {
    ...
    'update_locations': {
        'task': 'cron.tasks.update_locations',
        'schedule': crontab(hour='10', minute='0')
    },
    ...
}
```

at 10 UTC beat executes the task as expected

```
[2015-05-13 10:00:00,046: DEBUG/MainProcess] cron.tasks.update_locations sent. id->a1c53d0e-96ca-4673-9d03-972888581176
```

but this task never arrives to rabbitmq (I'm using the tracing module in rabbitmq to track incoming messages). I have several other tasks which seem to run fine with beat but certain tasks like the one above never run. Running the tasks manually in django with cron.tasks.update_locations.delay() runs the task with no problem.

**Comment by PMickael** [comment]
Try to set a optional queue
```python
CELERYBEAT_SCHEDULE = {
    ...
    'update_locations': {
        'task': 'cron.tasks.update_locations',
        'schedule': crontab(hour='10', minute='0'),
        'options': {'queue': 'test'}
    },
    ...
}
```
And let see if a task is create (do not set a worker on this queue)

**Comment by pimlinders** [comment]
@PMickael i'll give that a shot. I'll follow up once i've implemented this.

**Comment by thedrow** [comment]
Could it be related to https://github.com/celery/celery/issues/2620?

**Comment by pimlinders** [comment]
@thedrow In the celery beat logs I see no mention of "Tasks flagged as revoked" so I don't think so.

@PMickael I added the optional queue and did not see the task getting populated. 

I haven't been able to reproduce the problem locally. I'm setting up a new server to see if the issue persist.

**Comment by thedrow** [comment]
It might be the other way around since I'm experiencing this problem as well.

**Comment by thedrow** [comment]
@pimlinders Can you share the offending code or provide a test case I can work with?

**Comment by pimlinders** [comment]
I worked on this for quite some time and was not able to identify a single piece of code that was causing the issue. Seemed to occur randomly, some tasks ran and some didn't but then later would work. Beat always reported the tasks as being executed even though they were never received by rabbitmq.

I'm running celery with beat on an ec2 t2.micro ubunutu instance. Due the limited memory and beat  using roughly 350mb I ended up switching to crontab to execute custom django commands which called delayed tasks. Once I did ubuntu started randomly page faulting when executing tasks, essentially the same as what I was seeing with beat only now the OS was reporting an error. 

I eventually moved to Debian which solved all my problems. Must of been something wrong with the version of Ubuntu, installed libraries or the VM it self. I never got around to using beat again but my suspicion is that it must of been a bad VM. I'll close this now as I can't provide much more information.
